### PR TITLE
Search in Properly Placed Pages

### PR DESCRIPTION
**Pull request by planarvoid**
Add search functionality to Properly Placed Pages. Search is implemented in code in PageStore. I assume we might have an endpoint in future that hits directly API. 

I've added some tests for Search (PagesViewModel, PageStore). Testing coroutines has been non-trivial, I'm planning to write a post about it. 

I had to bump mockito version in order to test coroutines.

### CODE REVIEW

**Comment by 0nko** [comment]
The search result seems to be missing the headers with the page type from the design:

image

File: WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt [hunk 1]
```diff
@@ -47,12 +48,40 @@ class PagesViewModel
...
 
     fun start(site: SiteModel) {
         this.site = site
 
         clearSearch()
         reloadPagesAsync()
+        handleThrottledSearchAsync()
+    }
+
+    private fun handleThrottledSearchAsync() = launch(CommonPool) {
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt]: yep :-) I was looking around for a solution and found a click handler done with an infinite loop and it seemed quite neat. However, I agree it's probably an overkill. I'll rewrite it

**Comment by planarvoid** [comment]
@0nko I haven't noticed the headers! I'm on it

File: WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt [hunk 2]
```diff
@@ -84,8 +86,15 @@ class PagesViewModel
                 delay(200)
                 searchJob = null
                 if (isActive) {
-                    val result = pageStore.search(site, searchQuery)
-                            .map { Page(it.pageId.toLong(), it.title, null) }
+                    val result = pageStore.groupedSearch(site, searchQuery)
+                            .map { (status, results) ->
+                                listOf(Divider(1, status.toResource())) +
```
Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt]: Is the `Id` property necessary in the `Divider` class?

Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt]: it is but it shouldn't be 1! :-) we should be using a hashcode to identify divider instead

**Comment by 0nko** [comment]
Nice job! 🎉

**Pull request merged**